lib/internal/url.js

```diff
@@ -1444,11 +1433,7 @@ function toPathIfFileURL(fileURLOrPath) {
 }
 
 function NativeURL(ctx) {
-  Object.defineProperty(this, context, {
-    enumerable: false,
-    configurable: false,
-    value: ctx
-  });
+  this[context] = ctx;
```

Comment:
I am wondering if we can just remove this class and inline the calls

```js
const url = new NativeURL(ctx);
url[searchParams] = new URLSearchParams();
url[searchParams][context] = url;
```

with

```js
const url = Object.create(URL.prototype);
url[context] = ctx;
const params = new URLSearchParams();
params[context] = url;
url[searchParams] = params;
```

(Not sure about the performance implications, though, but it seems cleaner if we reduce one more class that looks like URL-related since this is only used by the C++ classes, and there are already many URL classes so it's a bit hard to find which one is the one you are looking for)